nixos/modules/services/web-apps/kavita.nix

```diff
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+
+let cfg = config.services.kavita;
+in {
+  options.services.kavita = {
+    enable = lib.mkEnableOption (lib.mdDoc "Kavita reading server");
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      default = "kavita";
+      description = lib.mdDoc "User account under which Kavita runs.";
+    };
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.kavita;
+      defaultText = "pkgs.kavita";
+      description = lib.mdDoc "Kavita package to use.";
+    };
+
+    dataDir = lib.mkOption {
+      default = "/var/lib/kavita";
+      type = lib.types.str;
+      description = lib.mdDoc "The directory where Kavita stores its state.";
+    };
+
+    tokenKeyFile = lib.mkOption {
+      type = lib.types.path;
+      description = lib.mdDoc ''
+        A file containing the TokenKey, a secret with at 128+ bits. One can be
+        generated with `head -c 32 /dev/urandom | base64`.
```

Comment:
```suggestion
        A file containing the TokenKey, a secret with at 128+ bits. 
        It can be generated with `head -c 32 /dev/urandom | base64`.
```